nullaway/src/main/java/com/uber/nullaway/ClassAnnotationInfo.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2017-2022 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.util.Context;
+
+/**
+ * Provides APIs for querying whether code is annotated for nullness checking, and for related
+ * queries on what annotations are present on a class and/or enclosing classes. Makes use of caching
+ * internally for performance.
+ */
+public final class ClassAnnotationInfo {
+
+  private static final Context.Key<ClassAnnotationInfo> ANNOTATION_INFO_KEY = new Context.Key<>();
+
+  private static final int MAX_CACHE_SIZE = 200;
+
+  private final Cache<Symbol.ClassSymbol, CacheRecord> cache =
+      CacheBuilder.newBuilder().maximumSize(MAX_CACHE_SIZE).build();
+
+  private ClassAnnotationInfo() {}
+
+  /**
+   * Get the ClassAnnotationInfo for the given javac context. We ensure there is one instance per
+   * context (as opposed to using static fields) to avoid memory leaks.
+   */
+  public static ClassAnnotationInfo instance(Context context) {
+    ClassAnnotationInfo cache = context.get(ANNOTATION_INFO_KEY);
```

Comment:
Very minor nit, but, can we rename the local here? It's no longer directly the cache (that's now... `cache.cache` based on this variable name :) )

Reply:
8d35f3164bc6d547dd4e3ec5d04d9037e71b40de